test/e2e/adminapi_kubernetesobjects.go

```diff
@@ -0,0 +1,350 @@
+		})
+
+		It("should not be able to delete a secret", func() {
+			By("deleting the secret via RP admin API")
+			params := url.Values{
+				"kind":      []string{"secret"},
+				"namespace": []string{namespace},
+				"name":      []string{objName},
+			}
+			var cloudErr api.CloudError
+			resp, err := adminRequest(ctx, resourceID, http.MethodDelete, "/kubernetesobjects", params, nil, &cloudErr)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(Equal(http.StatusForbidden))
+			Expect(cloudErr.Code).To(Equal(api.CloudErrorCodeForbidden))
+		})
+	}
+
+	testConfigMapCreateOK := func() {
+		By("creating a new object via RP admin API")
+		body := fmt.Sprintf(
+			`{
+				"kind": "ConfigMap",
+				"apiVersion": "v1",
+				"metadata": {
+					"name": "%s",
+					"namespace": "%s"
+				},
+				"data": {
+					"key": "value"
+				}
+			}`,
+			objName,
+			namespace,
+		)
+		resp, err := adminRequest(ctx, resourceID, http.MethodPost, "/kubernetesobjects", nil, json.RawMessage(body), nil)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+	}
+
+	testConfigMapGetOK := func() {
+		By("getting an object via RP admin API")
+		params := url.Values{
+			"kind":      []string{"configmap"},
+			"namespace": []string{namespace},
+			"name":      []string{objName},
+		}
+		var obj unstructured.Unstructured
+		resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &obj)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("comparing it to the actual object retrived via Kubernetes API")
+		cm, err := clients.Kubernetes.CoreV1().ConfigMaps(namespace).Get(objName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(obj.GetNamespace()).To(Equal(cm.Namespace))
+		Expect(obj.GetName()).To(Equal(cm.Name))
+		Expect(obj.GetAnnotations()).To(Equal(cm.Annotations))
+	}
+
+	testConfigMapListOK := func() {
+		By("requesting a list of objects via RP admin API")
+		params := url.Values{
+			"kind":      []string{"configmap"},
+			"namespace": []string{namespace},
+		}
+		var obj unstructured.Unstructured
+		resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &obj)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("comparing names from the list action response with names retrived via Kubernetes API")
+		objs, err := obj.ToList()
+		Expect(err).NotTo(HaveOccurred())
+		var names []string
+		for _, o := range objs.Items {
+			names = append(names, o.GetName())
+		}
+		Expect(names).To(ContainElement(objName))
+	}
+
+	testConfigMapUpdateOK := func() {
+		By("updating the object via RP admin API")
+		body := fmt.Sprintf(
+			`{
+				"kind": "ConfigMap",
+				"apiVersion": "v1",
+				"metadata": {
+					"name": "%s",
+					"namespace": "%s"
+				},
+				"data": {
+					"key": "value"
+				}
```

Comment:
this update doesn't appear to be changing anything in the object?
I'm actually surprised this works at all, I would expect you to need to populate the resourceVersion in the metadata?

Reply:
> this update doesn't appear to be changing anything in the object?

Good catch! It's a copy-paste issue: I missed it while trying to restructure the test to make it more readable.

> I'm actually surprised this works at all, I would expect you to need to populate the resourceVersion in the metadata?

`resourceVersion` is optional. If there is potential race condition and you want to make sure that you update the object, you can do the following:

- Get an object from API
- Update it and put it back to API keeping `resourceVersion` provided by API server
- Retry on conflict

But we do not include it as it is not important in this test.